tests/apps/btc_app/btc_txn_helpers_tests.c

```diff
@@ -400,6 +400,137 @@ TEST(btc_txn_helper_test, btc_txn_helper_verify_input_p2wpkh_fail) {
   TEST_ASSERT_EQUAL_INT(2, status);
 }
 
+
+TEST(btc_txn_helper_test, btc_txn_helper_verify_input_p2wpkh_in_p2sh) {
+  /* Test data source: Bip143
+   * https://blockstream.info/testnet/tx/b095225e60df2896a4e03fcc4fb2ba9622ee513f34b3e8de45c25a4793a244a2?expand
+   */
+  int status = 2;
+  uint8_t raw_txn[2000] = {0};
+  hex_string_to_byte_array(
+      "0200000000010258afb1ece76f01c24f4935f453d210518163cb1d0383eaec331b202ebeb5e389"
+      "0000000017160014a76cad25cb569bb47305513ebedd6011dc419deeffffffff2b3682b3592588"
+      "5001f0e321df28d4ac675a9cbbccef2a69533bea7c5e5ad2c40000000017160014a76cad25cb56"
+      "9bb47305513ebedd6011dc419deeffffffff02941100000000000017a914bd7aabdeeef211b1bd"
+      "ad7218e14fea6c032101c087f22f00000000000017a914eaf97514c5ac1e41e413502e97ae42eb"
+      "f27ace3a870247304402206e038f4712541d699697ed55efc41219df4f244fc72caa5edd653837"
+      "f6555f6f02201cd8ea15b65fda17992abafaed86e066c3271ac16b9c46c54c2192438843dd0401"
+      "21029f75e1ef6b04e004a308b1f59215a8a3a5b7958bbcf184cc24ba7ab6574448780248304502"
+      "2100d15ce61648edc28b8b5a3531b80a1e8fc3b3eebe7d3fc4ca962cb04afc770dda02207c7eaf8"
+      "82d7fac45d2752f20e48d2f896715cbc5a3b0f5de3e19fea0da99beac0121029f75e1ef6b04e004"
+      "a308b1f59215a8a3a5b7958bbcf184cc24ba7ab65744487800000000",
+      838,
+      raw_txn);
+  // only fill necessary values
+    btc_txn_input_t input[] = {
+        {
+        .value = 4500,
+        .prev_output_index = 0x00000000,
+        .script_pub_key = {.size = 23},
+        },
+        {
+        .value = 12274,
+        .prev_output_index = 0x00000001,
+        .script_pub_key = {.size = 23},
+        }
+    };
+
+  hex_string_to_byte_array(
+      "b095225e60df2896a4e03fcc4fb2ba9622ee513f34b3e8de45c25a4793a244a2",
+      64,
+      input[0].prev_txn_hash);
+  // revere order of txn-id:
+  // A2 44 A2 93 47 5A C2 45 DE E8 B3 34 3F 51 EE 22 96 BA B2 4F CC 3F E0 A4 96 28 DF 60 5E 22 95 B0
+  cy_reverse_byte_array(input[0].prev_txn_hash, sizeof(input[0].prev_txn_hash));
+
+  hex_string_to_byte_array(
+      "b095225e60df2896a4e03fcc4fb2ba9622ee513f34b3e8de45c25a4793a244a2",
+      64,
+      input[1].prev_txn_hash);
+  // revere order of txn-id:
```

Comment:
reverse spellcheck

Reply:
My bad. Copied the comments from another section of the code